File: LeetCode/050.powx-n/powx-n.java
class Solution {
    public double myPow(double x, int n) {
        long ln = n;
        boolean isNegative = false;
        if (n < 0) {
            ln = -ln;
            isNegative = true;
        }
        double res = 1;
        while (ln != 0){
            if ((ln & 1) != 0){
                res *= x;
            }
            x *= x;
            ln >>= 1;
        }
        if (isNegative){
            res = 1 / res;
        }
        return res;
    }
}

File: LeetCode/0050.powx-n/powx-n.cpp
class Solution {
public:
    double myPow(double x, int n) {
        bool isNegative = false;
        unsigned int un;
    	if (n < 0){
            isNegative = true;
        	un = -n;
        }else{
        	un = n;
        }
        cout << un << endl;
    	double res = 1;
    	while (un){
    		if (un & 1){
    			res *= x;
    		}
    		x *= x;
    		un >>= 1;
    	}
        if (isNegative){
            res = 1 / res;
        }
    	return res;
    }
};